### PR TITLE
Add counter to Index and count the number of times the index was used

### PR DESCRIPTION
**Pull request by Avital-Fine**
This PR adds a new field to the FT.INFO command. 
The field is called `num_of_uses` and counts the number of calls to the index (ft.create, ft.info, etc).

When we move to multi readers this counter needs to be atomic.

### CODE REVIEW

File: tests/pytests/test.py [hunk 1]
```diff
@@ -3515,6 +3515,16 @@ def test_free_resources_on_thread(env):
...
 
+def testUsesCounter(env):
+    env.expect('ft.create', 'idx', 'ON', 'HASH', 'NOFIELDS', 'schema', 'title', 'text').ok()
+    env.execute_command('ft.info', 'idx')
+    env.execute_command('ft.search', 'idx', '*')
+
+    res = env.execute_command('ft.info', 'idx')
+    d = {res[i]: res[i + 1] for i in range(0, len(res), 2)}
+
+    env.assertEqual(str(d['number_of_uses']), '3')
```
Comment (ashtul) [inline tests/pytests/test.py:3526]: Lets use `assertInfoField`.

File: coord/src/info_command.c [hunk 2]
```diff
@@ -32,7 +32,8 @@ static InfoFieldSpec toplevelSpecs_g[] = {
     {.name = "offset_bits_per_record_avg", .type = InfoField_DoubleAverage},
     {.name = "indexing", .type = InfoField_WholeSum},
     {.name = "percent_indexed", .type = InfoField_DoubleAverage},
-    {.name = "hash_indexing_failures", .type = InfoField_WholeSum}};
+    {.name = "hash_indexing_failures", .type = InfoField_WholeSum},
+    {.name = "number_of_uses", .type = InfoField_Max}};
```
Comment (ashtul) [inline coord/src/info_command.c:36]: @K-Jo WDYT?

**Review by ashtul — COMMENTED** [review]
Looks nice.

Small comments.

File: coord/src/info_command.c (hunk 2 above)
Comment (oshadmi) [reply coord/src/info_command.c:36]: We can also consider to count reads and writes separately, e.g., `FT.SEARCH` is a read operation and `ALIASADD` is a write operation (`number_of_reads`, `number_of_writes`)

I think we should not count DEBUG command usages, e.g., TTL

Comment (oshadmi) [reply coord/src/info_command.c:36]: What about adding last access time (timestamps), so for example, in `INFO` command we could ask for the 5 most used indices, and the 5 most recently used indices.

In addition, I suggest to commit massive formatting changes in a separate commit than actual code changes.

**Comment by MeirShpilraien** [comment]
@Avital-Fine lets extend the top comment to include the API changes (the new fields and its meaning). Also I believe this change requires also to support it on the coordinator.

File: coord/src/info_command.c (hunk 2 above)
Comment (Avital-Fine) [inline coord/src/info_command.c:36]: @MeirShpilraien Is this what you mean by supporting in the coordinator?

**Comment by rafie** [comment]
@Avital-Fine We can say something in the docs (i.e. `docs/commands/ft.info.md`) about it.

**Review by ashtul — APPROVED** [review]
Nice job!

**Pull request merged**